connectivity/mqtt/duplex_callback.c

```diff
@@ -110,7 +111,7 @@ static status_t mqtt_request_handler(mosq_config_t *cfg, char *subscribe_topic,
 
   // Set response publishing topic with the topic we got message and the Device ID (client ID) we got in the message
   int res_topic_len = strlen(subscribe_topic) + 1 + ID_LEN + 1;
-  char *res_topic = (char *)malloc(res_topic_len);
+  res_topic = (char *)malloc(res_topic_len);
```

Comment:
You don't have to change line. Minimize each functional changes as possible.

Reply:
If we don't move the declaration of `res_topic` to the beginning, then we will meet SEGV fault at line 130 and which is the memory leak we aimed to fix in this PR.

Reply:
The SEGV fault happened when we meet errors before line 114, which would trigger jumping the `done` label, and at this moment, `res_topic` has not been initialized.

Reply:
Got it. I'm thinking of the integration for smart pointer techniques: https://snai.pe/posts/c-smart-pointers